app/src/org/commcare/CommCareApp.java

```diff
@@ -411,6 +414,17 @@ public SQLiteDatabase getHandle() {
         };
     }
 
+    public PrimeEntityCacheHelper getPrimeEntityCacheHelper() {

```

Comment:
Thanks for the refactor, I think this coupling will make a lot of sense and keep things straight